### DISCUSSION TITLE
Avoid `@CreateRequestContext()` in NestJS

### DISCUSSION THREAD

**Discussion by Borderliner** [mikro-orm/mikro-orm#5561] [Q&A]
I want to avoid using `@CreateRequestContext()` on every route because I'm lazy. Is it possible to auto-fork with middleware in NestJS? I tried adding `MikroOrmMiddleware` to `app.module.ts` to no avail like this:
```ts
export class AppModule implements NestModule {
    configure(consumer: MiddlewareConsumer) {
        consumer.apply(MikroOrmMiddleware).forRoutes('*')
    }
}
```
I don't know if it makes any difference, but notice that I'm initializing `MikroOrmModule` in another `DatabaseModule` and not `AppModule` itself. `AppModule` has dependency on `DatabaseModule` where it uses `MikroOrmModule.forRoot()`.

I wanted to use the below in my `main.ts` file, but didn't know how to initialize the `orm` variable directly.
```ts
app.use((req, res, next) => {
  RequestContext.create(orm.em, next)
})
```

**Comment by B4nan** [comment]
You are not supposed to do that, the decorator is there for a very specific use case (e.g. cron jobs), for HTTP request handles the nest adapter registers a middleware automatically.

> I wanted to use the below in my main.ts file, but didn't know how to initialize the orm variable directly.

You don't, you need to get the value from nest DI if you want this approach. You should not need this either, it happens automatically.

**Reply by Borderliner** [reply]
The documentation on this is very vague. When I get the value from Nest DI it complains about using global context. I had to resort to `app.use` in my `main.ts`, a solution I don't really like.

**Reply by B4nan** [reply]
Do you use api prefix? There is a bug in nest with that, unforunatly they ignored me when i was reporting it. 

https://github.com/nestjs/nest/pull/13337

**Reply by Borderliner** [reply]
Yes I use api prefixes.

**Reply by B4nan** [reply]
If you have some spare time, would be great to create isolated reproduction (with just nest, no ORM involved - this is only about them not calling middleware properly with the API prefixing set up) and open a new issue on nest side, I'd say that's the only way to attract their attention.
